client/src/views/Models/components/Graphs/GlobalGraph.vue

```diff
@@ -118,12 +121,34 @@
         this.renderer.adapter = new DagreAdapter(DEFAULT_RENDERING_OPTIONS);
       }
 
-      const nodesHierarchy = hierarchyFn(this.data?.nodes); // Transform the flat nodes structure into a hierarchical one
+      // Transform the flat nodes structure into a hierarchical one
+      const nodesHierarchy = hierarchyFn(data.nodes);
       constructRootNode(nodesHierarchy); // Parse the data to a format that the graph renderer understands
-      const data = { nodes: [nodesHierarchy], edges: this.data?.edges };
+      data = { nodes: [nodesHierarchy], edges: data.edges };
 
       this.renderer.setData(data);
-      return this.renderer.render();
+      await this.renderer.render();
+
+      // Collapse top-level boxes by default
+      // HACK: The collapse/expand functions are asynchronous and trying to execute them all at once
+      // seems to create problems with the tracker.
+      const collapsedIds = calcNodesToCollapse(this.layout, this.renderer.layout);
+      if (collapsedIds.length > 0) {
+        const collapseNodeTimeOut = (nextID, renderer) => {
+          return new Promise((resolve) => {
+            setTimeout(() => {
+              renderer.collapse(nextID);
+              resolve();
+            }, 500);
+          });
+        };
+
+        collapsedIds.reduce((accumulatorPromise, nextID) => {
+          return accumulatorPromise.then(() => {
+            return collapseNodeTimeOut(nextID, this.renderer);
+          });
+        }, Promise.resolve());
+      }
```

Comment:
this whole section can be replaced by the following:

```
      collapsedIds.map((nextId, i) => {
        setTimeout(() => this.renderer.collapse(nextId), i * 500);
      });
```

regarding the hack, i looked and it seems like there is no other way to do this. the way to fix this might be to have the `collapse` function return a promise when its safe to call collapse again. that being said, currently the `collapse` function is already returning a promise, but that promise does not seem to be helpful in knowing when calling it again is ok. could this be a bug?

Reply:
also, is 500 ms necessary? can this (or should this) be shortened?

Reply:
Careful as `Array.map()` returns an array, in this case `Array.forEach()` would be more appropriate.

Reply:
I talked to Daniel Chang (from WM) about this and he mentioned that we might be able to remove the `async` nature of the `collapse/expand` functions. I think we can try to merge this PR now and then, explore how to change that in the original svg-flowgraph library.

Reply:
I tried to shorten the time to 100ms but I get an error. I'm going to keep it like that and we can see if we can remove it later.